### DISCUSSION TITLE
Przepisałem kod Smogomierza

### DISCUSSION THREAD

**Discussion by stsdc** [hackerspace-silesia/Smogomierz#169] [General]
Smogomierz to ważny projekt, które przez ostatnie kilka lat mocno się rozwinął, uzyskał sporo ważnych funkcjonalności i wsparcie dla wielu czujników. Jednak ostatnie wersje Smogomierza już miały spore problemy z poprawnym działaniem (przynajmnie u mnie). Na przykład 2.8.1 miała bootloopa, a w wersji 2.7.9 był problem z wyświetlaniem stron html.

Podjąłem próbę naprawienia kodu, jednak się skończyło na przepisaniu wszystkiego. :sweat_smile: 

Żeby w miare domknąć projekt musiałem mocno przyciąć funkcjonalności oraz wsparcie, ale daje to dobrą bazę dla dalszego rozwoju. 

* Płytka: ESP32
* Czujniki BME280, PMS7003
* Pomiar napięcia (do poprawy)
* Wysyłanie danych co 2 min
  - Sensors.Community (Luftdaten) 
  - aqi.eco
* Domyślnie włączony DeepSleep
* Przycisk *boot* wybudza płytkę oraz wyłącza DeepSleep aż do kolejnego restartu
* Strona ustawień:
   - Pokazuje ID
   - Zapisuje ścieżkę aqi.eco  
 * WIP: Wsparcie dla Firebase RTDB

Postanowiłem podzielić kodem projektu: być może ktoś będzie mógł go wykorzystać dla siebie, albo poprawić mój:
https://github.com/stsdc/smoggy

**Comment by hanskiepl** [comment]
Cześć, a działa tam MQTT czy zostało wycięte? Zastanawiam się jak te dane zaciągnąć sobie do Home Assistanta.

**Reply by stsdc** [reply]
Wycięte, ale można dodać ;)
